backend_python/services/vector_db.py

```
@@ -93,23 +110,19 @@ def embed_function(texts):
+            # For ChromaDB 0.3.23: Create collection WITHOUT embedding function
+            # We generate embeddings manually to avoid the collection.add() bug
+            # This is the most stable approach used in production
             try:
-                self.collection = self.client.get_collection(
-                    name=self.collection_name,
-                    embedding_function=embedding_function
-                )
-                logger.info(f"Connected to existing '{self.collection_name}' collection")
+                self.collection = self.client.get_collection(name=self.collection_name)
+                logger.info(f"Using existing '{self.collection_name}' collection")
             except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)